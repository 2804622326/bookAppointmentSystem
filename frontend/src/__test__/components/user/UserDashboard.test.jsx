import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import UserDashboard from '../../../components/user/UserDashboard.jsx';

jest.mock('../../../components/user/UserService.js', () => ({
  getUserById: jest.fn(() => Promise.resolve({ data: { id: '1', userType: 'PATIENT', appointments: [], reviews: [] } })),
  deleteUser: jest.fn(() => Promise.resolve({ message: 'deleted' }))
}));

jest.mock('../../../components/modals/ImageUploaderService.jsx', () => ({
  deleteUserPhoto: jest.fn(() => Promise.resolve({ message: 'ok' }))
}));

test('renders dashboard and stores tab key', async () => {
  localStorage.setItem('userId', '1');
  const { asFragment } = render(
    <MemoryRouter initialEntries={['/user-dashboard/1/my-dashboard']}>
      <Routes>
        <Route path='/user-dashboard/:userId/my-dashboard' element={<UserDashboard />} />
      </Routes>
    </MemoryRouter>
  );
  await waitFor(() => screen.getByText(/profile/i));
  expect(asFragment()).toMatchSnapshot();
});
