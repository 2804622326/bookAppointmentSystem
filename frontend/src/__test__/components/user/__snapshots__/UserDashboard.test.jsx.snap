// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dashboard and stores tab key 1`] = `
<DocumentFragment>
  <div
    class="mt-2 user-dashboard container"
  >
    <ul
      class="mb-2 nav nav-tabs nav-justified"
      role="tablist"
    >
      <li
        class="nav-item"
        role="presentation"
      >
        <button
          aria-controls="react-aria-:r0:-tabpane-profile"
          aria-selected="true"
          class="nav-link active"
          data-rr-ui-event-key="profile"
          id="react-aria-:r0:-tab-profile"
          role="tab"
          type="button"
        >
          <h3>
            Profile
          </h3>
        </button>
      </li>
      <li
        class="nav-item"
        role="presentation"
      >
        <button
          aria-controls="react-aria-:r0:-tabpane-status"
          aria-selected="false"
          class="nav-link"
          data-rr-ui-event-key="status"
          id="react-aria-:r0:-tab-status"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <h3>
            Appointments Overview
          </h3>
        </button>
      </li>
      <li
        class="nav-item"
        role="presentation"
      >
        <button
          aria-controls="react-aria-:r0:-tabpane-appointments"
          aria-selected="false"
          class="nav-link"
          data-rr-ui-event-key="appointments"
          id="react-aria-:r0:-tab-appointments"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <h3>
            Appointment Details
          </h3>
        </button>
      </li>
      <li
        class="nav-item"
        role="presentation"
      >
        <button
          aria-controls="react-aria-:r0:-tabpane-reviews"
          aria-selected="false"
          class="nav-link"
          data-rr-ui-event-key="reviews"
          id="react-aria-:r0:-tab-reviews"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <h3>
            Reviws
          </h3>
        </button>
      </li>
    </ul>
    <div
      class="tab-content"
    >
      <div
        aria-labelledby="react-aria-:r0:-tab-profile"
        class="fade tab-pane active show"
        id="react-aria-:r0:-tabpane-profile"
        role="tabpanel"
      >
        <div
          class="col"
        >
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="col-md-3"
              >
                <div
                  class="text-center mb-3 shadow card"
                >
                  <div
                    class="card-body"
                  >
                    <img
                      alt="User photo"
                      class="card-img user-image"
                      src="test-file-stub"
                    />
                  </div>
                  <div
                    class="text-center"
                  >
                    <p>
                       
                      <a
                        href="/user-dashboard/1/my-dashboard"
                      >
                        Update Photo
                      </a>
                    </p>
                    <p>
                       
                      <a
                        href="/user-dashboard/1/my-dashboard"
                      >
                        Remove Photo
                      </a>
                    </p>
                    <p>
                       
                      <a
                        href="/user-dashboard/1/my-dashboard"
                      >
                        Change Password
                      </a>
                    </p>
                  </div>
                </div>
                <div
                  class="card-body"
                >
                  <div
                    class="d-flex justify-content-center mb-4"
                  >
                    <div
                      class="mx-2"
                    >
                      <a
                        class="btn btn-warning btn-sm"
                        href="/update-user/1/update"
                      >
                        Edit profile
                      </a>
                    </div>
                    <div
                      class="mx-2"
                    >
                      <button
                        class="btn btn-danger btn-sm"
                        type="button"
                      >
                        Close account
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="col-md-8"
              >
                <div
                  class="mb-3 shadow card"
                >
                  <div
                    class="d-flex align-items-center card-body"
                  >
                    <div
                      class="col-md-4"
                    >
                      First Name :
                    </div>
                    <div
                      class="col-md-4"
                    >
                      <p
                        class="card-text"
                      />
                    </div>
                  </div>
                  <div
                    class="d-flex align-items-center card-body"
                  >
                    <div
                      class="col-md-4"
                    >
                      Last name :
                    </div>
                    <div
                      class="col-md-4"
                    >
                      <p
                        class="card-text"
                      />
                    </div>
                  </div>
                  <div
                    class="d-flex align-items-center card-body"
                  >
                    <div
                      class="col-md-4"
                    >
                      Gender :
                    </div>
                    <div
                      class="col-md-4"
                    >
                      <p
                        class="card-text"
                      />
                    </div>
                  </div>
                  <div
                    class="d-flex align-items-center card-body"
                  >
                    <div
                      class="col-md-4"
                    >
                      Email :
                    </div>
                    <div
                      class="col-md-4"
                    >
                      <p
                        class="card-text"
                      />
                    </div>
                  </div>
                  <div
                    class="d-flex align-items-center card-body"
                  >
                    <div
                      class="col-md-4"
                    >
                      Mobile :
                    </div>
                    <div
                      class="col-md-4"
                    >
                      <p
                        class="card-text"
                      />
                    </div>
                  </div>
                  <div
                    class="d-flex align-items-center card-body"
                  >
                    <div
                      class="col-md-4"
                    >
                      User Type :
                    </div>
                    <div
                      class="col-md-4"
                    >
                      <p
                        class="card-text"
                      >
                        PATIENT
                      </p>
                    </div>
                  </div>
                  <div
                    class="d-flex align-items-center card-body"
                  >
                    <div
                      class="col-md-4"
                    >
                      Account status : 
                    </div>
                    <div
                      class="col-md-4"
                    >
                      <p
                        class="inactive card-text"
                      >
                        Inactive
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="mb-3 shadow card"
                >
                  <div
                    class="d-flex align-items-center card-body"
                  >
                    <div
                      class="col-md-2"
                    >
                      Role(s) :
                    </div>
                    <div
                      class="col-md-4"
                    >
                      <div
                        class="list-group list-group-flush"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="react-aria-:r0:-tab-status"
        class="fade tab-pane"
        id="react-aria-:r0:-tabpane-status"
        role="tabpanel"
      >
        <div
          class="row"
        >
          <h4
            class="text-center mt-4"
          >
            Appointment Overview
          </h4>
          <div
            class="col"
          >
            <div
              class="text-center mt-5"
            >
              <h5>
                No appointment data available at the moment
              </h5>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="react-aria-:r0:-tab-appointments"
        class="fade tab-pane"
        id="react-aria-:r0:-tabpane-appointments"
        role="tabpanel"
      >
         
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="text-center mt-5"
            >
              <h5>
                No appointment data available at the moment
              </h5>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="react-aria-:r0:-tab-reviews"
        class="fade tab-pane"
        id="react-aria-:r0:-tabpane-reviews"
        role="tabpanel"
      >
        <div
          class="d-flex justify-content-center align-items-center container"
        >
          <div
            class="mt-5 mb-4 review-card card"
          >
            <h4
              class="text-center mb-2"
            >
              Your Reviews
            </h4>
            <hr />
            <div
              class="row"
            >
              <div
                class="col"
              >
                <div
                  class="text-center mt-5"
                >
                  <h5>
                    No review data available at the moment
                  </h5>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
